language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -935,6 +957,22 @@ fn parse_cmd_(tokens: &mut Lexer) -> Result<Cmd_, ParseError<Loc, anyhow::Error>
             let (name, tys) = parse_name_and_type_actuals(tokens)?;
             parse_unpack_(tokens, name, tys)
         }
+        Tok::If => {
+            consume_token(tokens, Tok::If)?;
+            consume_token(tokens, Tok::LParen)?;
+            let cond = parse_exp(tokens)?;
+            consume_token(tokens, Tok::RParen)?;
+
+            let true_label = parse_block_label(tokens)?;
+            consume_token(tokens, Tok::Else)?;
+            let false_label = parse_block_label(tokens)?;
+
+            Ok(Cmd_::IfElse(Box::new(cond), true_label, false_label))
```

Comment:
We should change this to make the `else` branch optional. It would fix a lot of my issues with how the CFG changed in some of the tests

Reply:
I like this idea, but when I sat down to implement it I encountered a problem:

```
    let condition: bool;
    condition = false;
    if (condition) b1;
label b1:
    abort 1;
label b2:
    return;
```

This is related to my comment above about there no longer being a guarantee as to where the bytecode for the true and false blocks is generated: we need to decide on where to jump in the case that the `condition` is false. We could jump to the next label that appears after the true label -- so, we implicitly jump to `b2` in this case. However, doing so would be exactly the same as writing `if (condition) b1 else b2;`, just less explicit... or am I missing something?